WebPhapp/backend/server.js

```diff
@@ -981,30 +1142,207 @@ Returns:
         ...
     ]
 */
-app.get('/api/v1/dispensers/:name', (req, res) => {
-    var name = req.params.name;
+app.get('/api/v1/dispensers', (req, res) => {
+    var name = req.query.name; 
+    var finish = function(success, dispensers, error='') {
+        var msg;
+        if(success) {
+            msg = 'returning ' + dispensers.length.toString() + ' dispensers.';
+        } else {
+            msg = 'error: ' + error;
+        }
+        console.log('/api/v1/dispensers: ' + msg);
+        res.status(success ? 200 : 400).send(dispensers);
+    };
 
-    // if dispenser ID is null or undefined, return all
-    if(name == null) {
-        console.log('/api/v1/dispensers/prescriptions/:dispenserID: returning all dispensers');
-        res.status(200).send([]);
-        return;
+    // if no connection string (Travis testing), grab dispensers from json files
+    if (!conn.MySQL) {
+        var dispensers = readJsonFileSync(
+            __dirname + '/' + "dummy_data/dispensers.json").dispensers;
+
+        dispensers = dispensers.filter(function(elem) {
+            // if no query given, return all dispensers
+            if(name === undefined) return true;
+
+            // case insensitive: match substrings in dispenser name
+            return elem.name.toLowerCase().includes(name.toLowerCase());
+        });
+        return finish(true, dispensers);
     }
 
-    var dispensers = readJsonFileSync(
-        __dirname + '/' + "dummy_data/dispensers.json").dispensers;
+    // otherwise, grab dispensers from MySQL
+    if (name == undefined)  name = '';
+    mysql.getDispensersByName(name, connection)
+    .then((answer) => {
+        var dispensers = [];
+        for (var i = 0; i < answer.rows.length; i++){
+            dispensers.push({
+                dispenserID: answer.rows[i].id,
+                name: answer.rows[i].name,
+                location: answer.rows[i].location,
+                phone: answer.rows[i].phone
+            });
+        }
+        return finish(true, dispensers);
+    })
+    .catch((error) => {
+        return finish(false, [], error);
+    });
+});
 
-    dispensers = dispensers.filter(function(elem) {
-        // if no query given, return all dispensers
-        if(name === undefined) return true;
+// ------------------------
+//        Prescriber
+// ------------------------
+
+/*
+About:
+    An api endpoint that returns all prescribers.
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/prescribers/all"
+    To be used in Axois call:
+        .get("/api/v1/prescribers/all")
+Returns:
+    [
+        {
+            prescriberID (int),
```

Comment:
nit: also need to update comments in name queries to include first and last name fields :)